pkg/syncutil/stats_reporter.go

```diff
@@ -180,14 +179,7 @@ type Reporter struct {
 
 // NewStatsReporter creates a reporter for sync metrics.
 func NewStatsReporter() (*Reporter, error) {
-	r := &Reporter{now: now}
-	return r, r.registerCallback()
-}
-
-func (r *Reporter) registerCallback() error {
```

Comment:
nit/style: please correct me if I'm wrong but now it seems we have two styles of RegisteringCallbacks: one by doing `meter.RegisterCallback` and another by registering the callback func on the Gauge itself with a `WithXCallback` option -- it would be nice to pick one and be consistent.

Reply:
Updated to use `WithXCallBack` everywhere.